clients/drcachesim/tools/invariant_checker.cpp

```diff
@@ -604,6 +604,24 @@ invariant_checker_t::parallel_shard_memref(void *shard_data, const memref_t &mem
 
     if (memref.marker.type == TRACE_TYPE_MARKER &&
         memref.marker.marker_type == TRACE_MARKER_TYPE_TIMESTAMP) {
+#ifdef X86_32
+        // i#5634: Truncated for 32-bit, as documented.
+        // A 32 bit timestamp rolls over every 4294 seconds, so it needs to be
+        // considered when timetsamps are compared. The check assumes two
+        // consecutive timestamps will never be more than 2^31 microseconds (
+        // 2147 seconds) apart.
```

Comment:
nit: Normally the opening paren would be on the next line: doesn't seem a normal line break point

Reply:
fixed.